lib/command/Command.js

```diff
@@ -164,73 +164,52 @@ class Command {
     }
 
     permissionCheck(msg) {
-        let req = false;
-        if(this.requirements.custom && typeof this.requirements.custom === "function") {
-            req = true;
-            if(this.requirements.custom(msg)) {
-                return true;
-            }
+        if(this.requirements.custom && typeof this.requirements.custom === "function" && !this.requirements.custom(msg)) {
+            return false;
         }
-        if(typeof this.requirements.userIDs === "function") {
-            req = true;
-            if(this.requirements.userIDs(msg).includes(msg.author.id)) {
-                return true;
-            }
-        } else if(this.requirements.userIDs.length > 0) {
-            req = true;
-            if(this.requirements.userIDs.includes(msg.author.id)) {
-                return true;
+        if (this.requirements.userIDs) {
+            const userIds = typeof this.requirements.userIDs === "function" ? this.requirements.userIDs(msg) : this.requirements.userIDs;
+            if(userIds.length > 0 && !this.requirements.userIDs.includes(msg.author.id)) {
+                return false;
             }
         }
         if(!msg.channel.guild) {
-            return !this.guildOnly && !req;
+            if (this.guildOnly) {
+                return false;
+            }
         } else if(this.dmOnly) {
```

Comment:
I would rearrange this `if` to avoid the negated condition, makes it a bit clearer:

```js
if (msg.channel.guild) {
	if (this.dmOnly) {
		return false;
	}
} else if (this.guildOnly) {
	return false;
}
```